runner_manager/backend/gcloud.py

```diff
@@ -0,0 +1,175 @@
+import time
+from typing import List, Literal
+
+from google.api_core.exceptions import NotFound
+from google.api_core.extended_operation import ExtendedOperation
+from google.cloud.compute import (
+    Image,
+    ImagesClient,
+    Instance,
+    InstancesClient,
+    Operation,
+    ZoneOperationsClient,
+)
+from pydantic import Field
+
+from runner_manager.backend.base import BaseBackend
+from runner_manager.models.backend import Backends, GCPConfig, GCPInstanceConfig
+from runner_manager.models.runner import Runner
+
+
+class GCPBackend(BaseBackend):
+    name: Literal[Backends.gcloud] = Field(default=Backends.gcloud)
+    config: GCPConfig
+    instance_config: GCPInstanceConfig
+
+    @property
+    def compute_client(self) -> InstancesClient:
+        """Returns a GCP Compute Engine client."""
+        return InstancesClient()
+
+    @property
+    def image_client(self) -> ImagesClient:
+        """Returns a GCP Image client."""
+        return ImagesClient()
+
+    @property
+    def zone_operation_client(self) -> ZoneOperationsClient:
+        """Returns a GCP Zone Operation client."""
+        return ZoneOperationsClient()
+
+    def configure_instance(self, runner: Runner):
```

Comment:
This function should not exist anymore most of those operation and variable definition can be set as default value of the GCPInstanceConfig, you should be able to pass GCPInstanceConfig to `instance_resource` parameter.

Check if you can inherit from it `class GCPInstanceConfig(InstanceConfig, Instance):` and override some of the var, or define as default the one you need.